File: Medium/34. Find First and Last Position of Element in Sorted Array/searchRange.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        vector<int> ans;
        while(l <= r){
            int m = l + (r - l) / 2;
            if(nums[m] < target){
                l = m + 1;
            }else if(nums[m] > target){
                r = m - 1;
            }else if(nums[m] == target){
                r = m - 1;
            }
        }
        if(l >= nums.size() || nums[l] != target){
            ans.push_back(-1);
        }else{
            ans.push_back(l);
        }

        l = 0;
        r = nums.size() - 1;
        while(l <= r){
            int m = l + (r - l) / 2;
            if(nums[m] < target){
                l = m + 1;
            }else if(nums[m] > target){
                r = m - 1;
            }else if(nums[m] == target){
                l = m + 1;
            }
        }
        if(r < 0 || nums[r] != target){
            ans.push_back(-1);
        }else{
            ans.push_back(r);
        }
        return ans;
    }
};

File: Medium/34. Find First and Last Position of Element in Sorted Array/searchRange.java
class Solution {
    int [] ans;
    public int[] searchRange(int[] nums, int target) {
        ans = new int [2];
        getLeft(nums, target);
        getright(nums, target);
        return ans;
    }
    private void getLeft (int [] nums, int target) {
        int len = nums.length;
        int left = 0;
        int right = len - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] ==  target) {
                right = mid - 1;
            }else if (nums[mid] < target) {
                left = mid + 1;
            }else if (nums[mid] > target) {
                right = mid - 1;
            }
        }
        if (left >= len || nums[left] != target) {
            ans[0] = -1;
        }else {
            ans[0] = left;
        }
    }
    
    private void getright (int [] nums, int target) {
        int len = nums.length;
        int left = 0;
        int right = len - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] ==  target) {
                left = mid + 1;
            }else if (nums[mid] < target) {
                left = mid + 1;
            }else if (nums[mid] > target) {
                right = mid - 1;
            }
        }
        if (right < 0 || nums[right] != target) {
            ans[1] = -1;
        }else {
            ans[1] = right;
        }
    }
}